github_slack_bot/utils/slack.py

```diff
@@ -0,0 +1,145 @@
+import logging
+import re
+from slackclient import SlackClient
+from slackclient.client import SlackNotConnected
+
+# TODO: Allow log level to be passed in as an application parameter
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d:%(funcName)s] %(message)s",
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+RTM_READ_INTERVAL = 1  # Number of seconds to wait between each rtm_read() call
+EXCEPTION_DEFAULT_MSG = "Something unexpected occurred, see ERROR log and Traceback above"
+
+
+class SlackBot:
+    """The SlackBot listens to events from the Slack RTM API and creates GitHub issues
+    from messages that mention the bot's username.
+    """
+
+    # TODO: extend is_bot_message() to allow bot mentions anywhere in the message
+    @staticmethod
+    def is_bot_message(bot_user_id, message):
+        """Check if the message mentions the bot_user_id at
+        the start of the message and includes a body
+        :param bot_user_id: bot's user ID to search for
+        :param message: message sent by user
+        :return: True if bot is mentioned and message contains body, False otherwise
+        """
+        message_stripped = message.strip()
+        logger.debug("Checking if bot '{}' is mentioned in message: '{}'"
+                     .format(bot_user_id, message_stripped))
+        return message_stripped.startswith("<@{}> ".format(bot_user_id))
+
+    # TODO: give user ability to provide a GitHub repo in their message body
+    # TODO: add ability to add assignees,labels, projects, milestones to Issues
+    @staticmethod
+    def parse_bot_message(msg):
+        """Parse a message mentioning our bot
+        :param msg: message sent by user
+        :return: A dict with the issue title and body if they exist,
+        a dict with an error message otherwise
+                Example:
+                {"ok": True, "issue_title": "Issue title here", "issue_body": "Issue body here"}
+                {"ok": False, "error": "Error message goes here"}
+        """
+        msg_regex = "^<@(.+?)> (.*)"
+        matches = re.search(msg_regex, msg)
+        msg_body = matches.group(2).strip()
```

Comment:
I think this will raise an exception if there was no match.

Reply:
This should be taken care of since we check for the correct format in the is_bot_message() function before calling this one, but I went ahead an updated the logic here to make it more robust